### PR TITLE
Fix CSV multi-line translation

### LINKED ISSUE #574: Issue with mulirow tooltip translation strings
I've noticed an issue while working with #572.

If translator add invisible line break instead of `\n` Crowdin generates multiple rows for singe translation string.The rest of translation after line break is not properly loaded in-game.

I suggest split multirow tooltips and separate key modifier (combination Ctrl/Shift/Alt with click) translation from tooltip text because it confuses translators.

### PR DESCRIPTION
**Pull request by krzychu124**
Fixes #574 

I created line joining function which should properly handle following cases:
- single or multiple newlines,
- commas,
- text with quotation-marks

I've created some tests for validation and translation parsing.

### CODE REVIEW

File: TLM/TLM/UI/Localization/LookupTable.cs [hunk 1]
```diff
@@ -117,15 +133,58 @@ public class LookupTable {
...
+            string[] lines = sr.ReadToEnd().Split(new[] { "\n", "\r\n" }, StringSplitOptions.None);
+            return ValidateAndJoinLines(lines);
+        }
+
+        /// <summary>
+        /// Validates lines by joining separated incomplete strings with new line characters
+        /// </summary>
+        /// <param name="lines">Lines of translation file to validate</param>
+        /// <returns>Validated and joined collection of translation strings - one per row</returns>
+        private static string[] ValidateAndJoinLines(string[] lines) {
```
Comment (kvakvs) [inline TLM/TLM/UI/Localization/LookupTable.cs]: What is this?
All you needed to do is change read cell algorithm, that if a `"` is open, then `\n` should be handled as part of the string.

Comment (kvakvs) [reply TLM/TLM/UI/Localization/LookupTable.cs]: https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/blob/master/TLM/TLM/UI/Localization/LookupTable.cs#L164-L186

Comment (krzychu124) [reply TLM/TLM/UI/Localization/LookupTable.cs]: Wait... `ReadCsvCell` is processing single line not whole file Stream. A few lines earlier whole file stream was split on lines.

Comment (kvakvs) [reply TLM/TLM/UI/Localization/LookupTable.cs]: I'd rather have file read entirely as a single string, and then we already have everything we need to parse cells, just fix those algorithms handling open quotes. I do not like joining and backtracking logic, we do not need it here.

Comment (krzychu124) [reply TLM/TLM/UI/Localization/LookupTable.cs]: I'll change it later today :wink:

**Comment by aubergine10** [comment]
Love the tests!

Appveyor builds failing though.

**Comment by aubergine10** [comment]
Test Run Failed.
Total tests: 53
     Passed: 49
     Failed: 4
 Total time: 1.3252 Seconds
Command exited with code 1

**Comment by aubergine10** [comment]
Error Message:
   CollectionAssert.Contains failed. 
  Stack Trace:
     at TMUnitTest.Translation.CsvParser.TestMultiLineCellParse_CRLF() in C:\projects\cities-skylines-traffic-manager-president-edition-labs\TLM\TMPE.UnitTest\Translation\CsvParser.cs:line 187

**Comment by krzychu124** [comment]
It should work now as expected

**Pull request merged**